.gitmodules

```diff
@@ -51,3 +51,7 @@
 [submodule "src/doc/rust-by-example"]
 	path = src/doc/rust-by-example
 	url = https://github.com/rust-lang/rust-by-example
+[submodule "src/llvm-emscripten"]
+	path = src/llvm-emscripten
+	url = https://github.com/rust-lang/llvm
+	branch = rust-llvm-release-4-0-1
```

Comment:
Shouldn't this be an Emscripten branch? It seems odd that emscripten has `branch = ` where other submodules don't.

Reply:
Indeed! I figured that for now it's probably easiest to do the smallest delta possible which is to switch to the same version we have right now. I think if we upgrade in the future though we should try to depend directly on Emscripten's fork of LLVM.

As for the `branch =` I think that's because I passed a `-b` argument to `git submodule add`, but as far as I can tell this is never used when cloning (as the submodule has a revision). It *may* be used for shallow clones of submodules but I never got that working historically..